app/controllers/orders_controller.rb

```diff
@@ -114,6 +114,24 @@ def cancel
     redirect_to checkout_order_path, alert: 'Payment was cancelled'
   end
 
+  def cancel_order
+    if @order.deals.any? { |deal| deal.expiring_soon? }
+      flash[:notice] = "Order can't be cancelled 30 minutes before the deal ends"
+    else
+      payment_intent = Stripe::PaymentIntent.retrieve(@order.payments.successful.payment_intent)
```

Comment:
let's move all stripe related code in a service called stripe_manager

Reply:
done